cpp/src/components/trim.cu

```diff
@@ -0,0 +1,186 @@
+
+template <typename vertex_t>
+struct extract_one_core_t {
+  __device__ thrust::optional<thrust::tuple<vertex_t, vertex_t>> operator()(vertex_t src,
+                                                                            vertex_t dst,
+                                                                            bool src_one_core,
+                                                                            bool dst_one_core,
+                                                                            thrust::nullopt_t) const
+  {
+    return (src_one_core ==  true) && (dst_one_core == true)
+             ? thrust::optional<thrust::tuple<vertex_t, vertex_t>>{thrust::make_tuple(src, dst)}
+             : thrust::nullopt;
+  }
+};
+
+template <typename edge_t>
+struct is_one_or_greater_t {
+  __device__ bool operator()(edge_t core_number) const
+  {
+    return core_number >= edge_t{1} ? true : false;
+  }
+};
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+std::tuple<rmm::device_uvector<vertex_t>,
+           rmm::device_uvector<vertex_t>,
+           std::optional<rmm::device_uvector<weight_t>>>
+trim(raft::handle_t const& handle,
+       cugraph::graph_view_t<vertex_t, edge_t, false, multi_gpu> const& graph_view,
+       std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+       bool do_expensive_check)
+{
+
+    rmm::device_uvector<edge_t> core_numbers(graph_view.number_of_vertices(), handle.get_stream());
+
+    core_number(
+      handle, graph_view, core_numbers.data(), k_core_degree_type_t::OUT, size_t{1}, size_t{1});
+
+   auto out_one_core_first =
+      thrust::make_transform_iterator(core_numbers.begin(), is_one_or_greater_t<edge_t>{});
+
+    rmm::device_uvector<uint8_t> out_one_core_flags(core_numbers.size(), handle.get_stream());
+
+    thrust::copy(handle.get_thrust_policy(),
+                 out_one_core_first,
+                 out_one_core_first + core_numbers.size(),
+                 out_one_core_flags.begin());
+
+    edge_src_property_t<decltype(graph_view), uint8_t> edge_src_out_one_cores(handle,
+                                                                                 graph_view);
+    edge_dst_property_t<decltype(graph_view), uint8_t> edge_dst_out_one_cores(handle,
+                                                                                 graph_view);
+    update_edge_src_property(
+      handle, graph_view, out_one_core_flags.begin(), edge_src_out_one_cores);
+    update_edge_dst_property(
+      handle, graph_view, out_one_core_flags.begin(), edge_dst_out_one_cores);
+
+    rmm::device_uvector<size_t> d_subgraph_offsets_out(2, handle.get_stream());
+    std::vector<size_t> h_subgraph_offsets_out{{0, out_one_core_flags.size()}};
+    raft::update_device(d_subgraph_offsets_out.data(),
+                      h_subgraph_offsets_out.data(),
+                      h_subgraph_offsets_out.size(),
+                      handle.get_stream());
```

Comment:
You need to extract a subgraph here and create a new graph from the extracted edges.